besu/src/main/scripts/unixStartScript.txt

```diff
@@ -182,15 +182,15 @@ APP_ARGS=`save "\$@"`
 # Collect all arguments for the java command, following the shell quoting and substitution rules
 eval set -- \$DEFAULT_JVM_OPTS \$JAVA_OPTS \$${optsEnvironmentVar} <% if ( appNameSystemProperty ) { %>"\"-D${appNameSystemProperty}=\$APP_BASE_NAME\"" <% } %>-classpath "\"\$CLASSPATH\"" <% if ( mainClassName.startsWith('--module ') ) { %>--module-path "\"\$MODULE_PATH\"" <% } %>${mainClassName} "\$APP_ARGS"
 
-unset BESU_USING_JEMALLOC
+#unset BESU_USING_JEMALLOC
```

Comment:
@amsmota I believe the original intent of this section of code is to detect availability of jemalloc library on the system and then set BESU_USING_JEMALLOC variable automatically. In other words, after you commented out the line 185 and 193, all of this section becomes redundant. 

What was the original problem that you have observed and trying to solve?

Reply:
In fact, regardless of the environment variable is present or not (or calculated by the `unixStartScript` or not), The Besu code will always attempt to detect jemalloc for logging purposes. 

https://github.com/hyperledger/besu/blob/8ca7129b0be3ba26b60ceb8920c01f717d5f5aca/besu/src/main/java/org/hyperledger/besu/cli/ConfigurationOverviewBuilder.java#L431

As I see it, the code in `detectJemalloc` should be simplified. The main purpose of `detectJeMalloc` is to simply **log** that the jemalloc library is being used by Besu Java process. The main issue that I see is our `unixStartScript` is the setting of `LD_PRELOAD`, it is possible that it is not detecting the correct path. 

This is why I am interested to know exactly what issue you are facing?

Reply:
Note that the purpose of environment variable `BESU_USING_JEMALLOC` is not to enable/disable using jemalloc. It is mainly used as an environment variable controlling the informational logging when Besu starts. The main variable that controls loading of jemalloc and make it available to Besu Java process is `LD_PRELOAD`.

Reply:
@fab-10 is my understanding about BESU_USING_JEMALLOC correct? ^

Reply:
correct

Reply:
Guys, thank you for reviwing and your comments. 

The intention of this PR is to be able for the developers or testers or DevOps people to be able to load or not load jemalloc based on the BESU_USING_JEMALLOC env. The motivation for this are:

- Developpers and Testers to be able to quickly compare the improvement in memory resources to be able to have a informed decision on using Jemalloc or another memory allocator or none at all.
- DevOps guys to be able to disable Jemalloc if the deployement fails. A real world scenario would be for the deployment to be sucessfull in DEV and UAT and fail in PROD (maybe because sligthly diferences in OS version or memory configuration or security settings, etc...). With this optional flag they can simply set BESU_USING_JEMALLOC to false and restart the deployment. Without it they would have no alternative to rollback the deployement and probably send that back to the dev team to remove Jemalloc altogether.

In relation to the BESU_USING_JEMALLOC flag, the intention is that it will be set at the operation system level, not in the script. That is why I commented out the lines in the unixStartScript. I left them there commented out there so reviwers could see what I did and eventually test for thenselves with or without it. 

The ConfigurationOverviewBuilder handles all the cases:

1) BESU_USING_JEMALLOC == true, it will load Jemalloc if it's not loaded yet:
ConfigurationOverviewBuilder 
```
                if (Boolean.parseBoolean(jemallocEnabled)) {
                  final String version = PlatformDetector.getJemalloc();
                  lines.add("jemalloc: " + version);
                } else {
                  logger.info(
                      "BESU_USING_JEMALLOC is present but is not set to true, jemalloc library not loaded");
                }
```
PlatformDetector
```
  public static String getJemalloc() {
    if (_jemalloc == null) {
      detectJemalloc();
    }

    return _jemalloc;
  }
```
```
 private static void detectJemalloc() {
    interface JemallocLib extends Library {...}

   *****************************************************************************
    final JemallocLib jemallocLib = Native.load("jemalloc", JemallocLib.class);
   *****************************************************************************

    PointerByReference pVersion = new PointerByReference();
    IntByReference pSize = new IntByReference(Native.POINTER_SIZE);
    jemallocLib.mallctl("version", pVersion, pSize, null, 0);

    _jemalloc = pVersion.getValue().getString(0);
}
```

2) BESU_USING_JEMALLOC == false, it will NOT load, it'll do nothing except
```
                  logger.info(
                      "BESU_USING_JEMALLOC is present but is not set to true, jemalloc library not loaded")
```
3) BESU_USING_JEMALLOC  does not exist at all, it will behave like the current implementatio (tries to load anyway)
```
              // in case the user is using jemalloc without BESU_USING_JEMALLOC env var
              try {
                final String version = PlatformDetector.getJemalloc();
                lines.add("jemalloc: " + version);
              } catch (final Throwable throwable) {
                logger.info(
                    "jemalloc library not found, memory usage may be reduced by installing it");
              }
```

I considered to change this last option and not load Jemalloc when the env var does not exist. It's probably safer but I decided to keep it compatible with the current impelmentation.

I tested this quite intensivelly with all 3 scenarios, I actually installed WSL on my Widnows machine to do it.

Please let me know what you think.

Cheers.

Reply:
Hi @amsmota, the requirements are clear, and I think the best way to implement them is to work on `besu-untuned` to completely remove the support for `jemalloc` from it, so you do not have any pre-configured tuning, and you can specify only the options you need, see my previous comment https://github.com/hyperledger/besu/pull/7424#issuecomment-2284458901

Reply:
> Note that the purpose of environment variable `BESU_USING_JEMALLOC` is not to enable/disable using jemalloc. It is mainly used as an environment variable controlling the informational logging when Besu starts. The main variable that controls loading of jemalloc and make it available to Besu Java process is `LD_PRELOAD`.

It is not so, in PlatformDetector::detectJemalloc it does load Jemalloc as I posted in my other msg...

Reply:
`PlatformDetector::detectJemalloc` is only needed to fetch the version of `jemalloc` on the overview, it makes no difference if that code is run or not.
The way `jemalloc` is used by the process is via `LD_PRELOAD`

Reply:
Yep, that's right, my bad... :(